api/v1beta1/externaldns_types.go

```diff
@@ -257,6 +257,13 @@ type ExternalDNSAWSProviderOptions struct {
 	// +kubebuilder:validation:Required
 	// +required
 	Credentials SecretReference `json:"credentials"`
+
+	// RoleARN contains the ARN of a IAM role that will be assumed when using the AWS API.
+	// It provides the ability to use a hosted zone in another AWS account.
+	//
+	// +kubebuilder:validation:Optional
```

Comment:
@alebedev87 Thoughts on adding validation? 

We did for CIO: https://github.com/openshift/api/blob/d976e23f87d346073608c0963d80f8a9e9d03d3a/config/v1/types_dns.go#L133

But, `--aws-assume-role` doesn't validate either. Is this something you think we should offer as a courtesy, or do you think it just makes it more fragile?

Reply:
I'm fine with the mentioned validation pattern. It's always better to catch problems earlier.

Reply:
Added validation.

Reply:
@alebedev87 switched validation from CRD to webhook and used `IsARN` https://github.com/openshift/external-dns-operator/blob/838b8fa38be41a4510ceafd255d780319b6a27c6/vendor/github.com/aws/aws-sdk-go/aws/arn/arn.go#L81.

Took idea from https://github.com/openshift/external-dns-operator/issues/190